src/module.js

```diff
@@ -0,0 +1,290 @@
+import _ from 'underscore';
+import Logger from './logger.js';
+import {capitalize, isNil} from 'lodash';
+import SpaceObject from './object.js';
+require('./lib/underscore-deep-extend-mixin.js');
+
+const Meteor = Meteor || undefined;
+const Npm = Npm || undefined;
+
+const Module = SpaceObject.extend('Space.Module',  {
+
+  ERRORS: {
+    injectorMissing: 'Instance of Space.Injector needed to initialize module.'
+  },
+
+  configuration: {},
+  requiredModules: null,
+  // An array of paths to classes that you want to become
+  // singletons in your application e.g: ['Space.messaging.EventBus']
+  // these are automatically mapped and created on `app.run()`
+  singletons: [],
+  injector: null,
+  _state: 'constructed',
+
+  Constructor(...args) {
+    SpaceObject.apply(this, args);
+    if (isNil(this.requiredModules)) {
+      this.requiredModules = [];
+    }
+  },
+
+  initialize(app, injector, isSubModule = false) {
+    this.app = app;
+    this.injector = injector;
+
+    if (!this.is('constructed')) {return;} // Only initialize once
+    if (isNil(this.injector)) {
+      throw new Error(this.ERRORS.injectorMissing);
+    }
+
+    this._state = 'configuring';
+
+    // Setup logger
+    if (!isSubModule) {
+      this.log = this._createLogger();
+    } else {
+      this.log = this.injector.get('log');
+    }
+    this.log.debug(`${this.constructor.publishedAs}: initialize`);
+
+    // Setup basic mappings required by all modules if this the top-level module
+    if (!isSubModule) {
+      this.injector.map('Injector').to(this.injector);
+      this._mapSpaceServices();
+    }
+
+    // Setup required modules
+    for (let moduleId of this.requiredModules) {
+      // Create a new module instance if not already registered with the app
+      if (isNil(this.app.modules[moduleId])) {
+        const ModuleClass = Module.require(moduleId, this.constructor.name);
+        this.app.modules[moduleId] = new ModuleClass();
+        // Initialize required module
+        const module = this.app.modules[moduleId];
+        module.initialize(this.app, this.injector, true);
+      }
+    }
+
+    // Merge in own configuration to give the chance for overwriting.
+    if (isSubModule) {
+      _.deepExtend(this.app.configuration, this.configuration);
+      this.configuration = this.app.configuration;
+    } else {
+      // The app can override all other modules
+      _.deepExtend(this.configuration, this.constructor.prototype.configuration);
+    }
+
+    // Provide lifecycle hook before any initialization has been done
+    if (_.isFunction(this.beforeInitialize)) {
+      this.beforeInitialize();
+    }
+
+    // @backward {space:base} <= 4.1.3 for Meteor package,
+    // Give every module access Npm
+    if (!isNil(Meteor) && this._isServer() && !isNil(Npm)) {
```

Comment:
I think this could be extracted to predefined dependencies. Such, would be added via static method on `Module.addPredefinedDependency/Module.setPredefinedDependencis`that would allow `space-base` be framework agnostic - however still, allow to add separate framework-oriented package with all sugary goodnes out of the box. 

This would result in Meteor adding default mappings like it was done previously here:  ```
  _mapMeteorApis: ->
    Space.log.debug("#{@constructor.publishedAs}: _mapMeteorApis")
   # Map Meteor standard packages
    @injector.map('Meteor').to Meteor
    if Package.ejson?
      @injector.map('EJSON').to Package.ejson.EJSON
    if Package.ddp?
      @injector.map('DDP').to Package.ddp.DDP
    if Package.random?
      @injector.map('Random').to Package.random.Random
    @injector.map('underscore').to Package.underscore._
    if Package.mongo?
      @injector.map('Mongo').to Package.mongo.Mongo
      if Meteor.isServer
        @injector.map('MongoInternals').to Package.mongo.MongoInternals

    if Meteor.isClient
      if Package.tracker?
        @injector.map('Tracker').to Package.tracker.Tracker
      if Package.templating?
        @injector.map('Template').to Package.templating.Template
      if Package.session?
        @injector.map('Session').to Package.session.Session
      if Package.blaze?
        @injector.map('Blaze').to Package.blaze.Blaze

    if Meteor.isServer
      @injector.map('check').to check
      @injector.map('Match').to Match
      @injector.map('process').to process
      @injector.map('Future').to Npm.require 'fibers/future'

      if Package.email?
        @injector.map('Email').to Package.email.Email

    if Package['accounts-base']?
      @injector.map('Accounts').to Package['accounts-base'].Accounts

    if Package['reactive-var']?
      @injector.map('ReactiveVar').to Package['reactive-var'].ReactiveVar```